### PR TITLE
src: Replace C cast

### PR DESCRIPTION
**Pull request by marioalexis84**
Replace C style casts. Mainly focused on FreeCAD and Coin classes.

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [ ]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [ ]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7053">pipeline status</a> for feature branch [PR_7053](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7053). Pipeline [567253980](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/567253980) was triggered at [ca3381e](https://github.com/FreeCAD/FreeCAD/pull/7053/commits/ca3381ea1f60d4121b54768bc4cfc086981740fa). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Review by chennes — APPROVED** [review]
This compiles and runs without issue on Linux and Windows, and is a welcome C++ modernization. I'll leave it open for another day or two in case anyone else wants to give it a look, since it does touch a lot of parts of the code.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7053">pipeline status</a> for feature branch [PR_7053](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7053). Pipeline [569720225](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/569720225) was triggered at [4e6a012](https://github.com/FreeCAD/FreeCAD/pull/7053/commits/4e6a012867143600e74cf875e36d8c1eb66a5410). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by marioalexis84** [comment]
@chennes  I polished the regex to search for the C casts a bit more.
Do you prefer that I add them to this same pull request or do you prefer the new cases found in a new pull request?
just so you don't have to revisit this PR again...

**Comment by chennes** [comment]
Go ahead and add to this one, I'll just re-review.

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7053">pipeline status</a> for feature branch [PR_7053](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7053). Pipeline [578985893](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/578985893) was triggered at [acf6a98](https://github.com/FreeCAD/FreeCAD/pull/7053/commits/acf6a98ac1b6636a71667da07e1c228bb0b8e38b). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Gui/Inventor/SoAutoZoomTranslation.h [hunk 1]
```diff
@@ -26,6 +26,13 @@
 #include <Inventor/fields/SoSFFloat.h>
 #include <Inventor/nodes/SoTransformation.h>
 
+class SoAction;
+class SoGetPrimitiveCountAction;
+class SoGetMatrixAction;
+class SoGLRenderAction;
+class SoGetBoundingBoxAction;
+class SoCallbackAction;
+class SoPickAction;
```
Comment (chennes) [inline src/Gui/Inventor/SoAutoZoomTranslation.h:35]: Why are the forward declarations needed here?

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7053">pipeline status</a> for feature branch [PR_7053](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7053). Pipeline [582756325](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/582756325) was triggered at [c153fb8](https://github.com/FreeCAD/FreeCAD/pull/7053/commits/c153fb82e5db3105db6bb1a6ceb7737f00c5167b). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by marioalexis84** [comment]
@chennes I have rebased this for when you can review again.

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7053">pipeline status</a> for feature branch [PR_7053](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7053). Pipeline [591602553](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/591602553) was triggered at [6599124](https://github.com/FreeCAD/FreeCAD/pull/7053/commits/6599124a1f22b819cd73512bdfae3e36e725f522). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by donovaly** [comment]
Looks good to me.
@chennes when you merge, I propose to keep the 14 commits just in case there is an issue (could also help when I get merge conflicts on backporting).

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7053">pipeline status</a> for feature branch [PR_7053](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7053). Pipeline [643207053](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/643207053) was triggered at [d191073](https://github.com/FreeCAD/FreeCAD/pull/7053/commits/d1910739802960d5e28a0a75d6cdde8a17f10cb8). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by marioalexis84** [comment]
@chennes ready to review.
other cast (int, float, etc) are pending for another pull request.
(although here I made a few).

**Comment by chennes** [comment]
Thanks for your patience, and sorry this fell off my radar in the first place. As @donovaly suggests above, I'm going to leave this un-squashed in case one of the commits causes issues individually, so it's easier to revert.

**Pull request merged**